### PR TITLE
3 integrate touchscreen calibration process

### LINKED ISSUE #3: Integrate Touchscreen Calibration Process

### PR DESCRIPTION
**Pull request by andiaye33**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -1,48 +1,230 @@
...
+
+    cal_coeff_x.a = cal_coeff_y.a = 1.0;
+    cal_coeff_x.b = cal_coeff_y.b = 0.0;
+
+     while (1) {
+        if (is_calibration_mode) {
+            lv_label_set_text(hello_world_label, "Calibration");
+
+            if (touch_point.pressed) {
+              printk("Calibration [%d]\n", point_index);
```
Comment (clementlgl) [inline src/main.c:203]: Same, shouldn't this be displayed in a log?

**Review by clementlgl — CHANGES_REQUESTED** [review]
Can you run pre-commit on all files? The code is not properly formatted.

File: src/main.c [hunk 2]
```diff
@@ -1,48 +1,230 @@
...
+            }
+        } else {
+            lv_label_set_text(hello_world_label, "6TRON BY CATIE!");
+            if (touch_point.pressed) {
+                touch_point.pressed = false;
+                draw_calibration_marker(touch_point.x, touch_point.y);
+            }
+        }
+
+        update_pwm();
```
[the remark this answers is not in the document]
Comment (Sokario) [reply src/main.c:225]: To be resolved in issue #5.

[the remark this answers is not in the document]
Comment (andiaye33) [reply src/main.c]: Will be resolved in another issues

**Comment by andiaye33** [comment]
@clementlgl : this PR is ready?

File: prj.conf [hunk 3]
```diff
@@ -1 +1,25 @@
...
+CONFIG_PWM=y
+
+CONFIG_LVGL=y
+CONFIG_LV_USE_LABEL=y
+
+# TOUCH SCREEN
+CONFIG_INPUT=y
+CONFIG_TSC2003_DEBOUNCE_MS=200
+CONFIG_INPUT_GPIO_KEYS=n
+CONFIG_INPUT_LOG_LEVEL_INF=y
```
Comment (clementlgl) [inline prj.conf:20]: I think this log is unnecessary

File: src/main.c [hunk 4]
```diff
@@ -1,48 +1,264 @@
...
+
+			if (touch_point.pressed) {
+				LOG_INF("Calibration [%d]\n", point_index);
+				touch_point.pressed = false;
+				calib_points.raw_x[point_index] = touch_point.x;
+				calib_points.raw_y[point_index] = touch_point.y;
+				point_index++;
+				if (point_index >= CALIBRATION_POINTS) {
+					is_calibration_mode = false;
+					// lv_label_set_text(cross_label, "");
```
Comment (clementlgl) [inline src/main.c:243]: Clean the code

File: src/main.c [hunk 5]
```diff
@@ -1,48 +1,264 @@
...
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/display.h>
 #include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/printk.h>
+#include <math.h>
```
Comment (clementlgl) [inline src/main.c:20]: All these includes are necessary? For example, I think `<zephyr/sys/printk.h>` is useless 🤔

**Review by clementlgl — CHANGES_REQUESTED** [review]

File: prj.conf [hunk 6]
```diff
@@ -1 +1,25 @@
+CONFIG_MAIN_STACK_SIZE=4096
+
+CONFIG_DISPLAY=y
+CONFIG_DISPLAY_LOG_LEVEL_ERR=y
```
Comment (Sokario) [inline prj.conf:4]: We don't need LOGs for display

File: prj.conf [hunk 7]
```diff
@@ -1 +1,25 @@
...
+
 CONFIG_GPIO=y
+
+CONFIG_LOG_DEFAULT_LEVEL=1
```
Comment (Sokario) [inline prj.conf:8]: Can you rename it to **LEVEL_ERR** ?

File: src/main.c [hunk 8]
```diff
@@ -1,48 +1,264 @@
...
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <inttypes.h>
+#include <lvgl.h>
 #include <stdio.h>
-#include <zephyr/kernel.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
```
Comment (Sokario) [inline src/main.c:11]: These includes includes each other, just pick the right one

**Review by Sokario — CHANGES_REQUESTED** [review]

**Review by clementlgl — APPROVED** [review]

**Comment by clementlgl** [comment]
@Sokario Can you review? It would be good to close this issue to move this example to a ‘samples’ repository.

**Review by Sokario — APPROVED** [review]

**Pull request merged**